### PR TITLE
Add auto-publishing and changelog generation.

### PR DESCRIPTION
**Pull request by jguiditta**
This was already added to most of our other projects, but toolchest did
not yet have this tooling.  Each commit should be published to
test-pypi, and on tag publish to pypi. Also generated the changelog as
we do elsewhere.  Patch includes a couple minor cleanups of related
tooling for consistency as well.

Signed-off-by: Jason Guiditta <jguiditt@redhat.com>

### CODE REVIEW

**Review by fuzzball81 — DISMISSED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Upload Package to Pypi and TestPyPI
+
+on:
+  push:
+    branches: [ main ]
```
Comment (jpichon) [inline .github/workflows/publish.yml:5]: This should be "master" I think, as that's the branch name here.

File: README.md [hunk 2]
```diff
@@ -55,24 +41,30 @@ for us, which you can activate via::
...
-See the `pipenv documentation <https://docs.pipenv.org/>`_ for more detail.
+See the [pipenv documentation](https://pipenv.pypa.io/en/latest/) for more detail.
+
+If you prefer to use pip directly in your venv, specify the following
+requirements files:
+
+  - requirements.txt
+  - test-requirements.txt
+
+There is also a dist-requirements.txt, if you will be building the project
```
Comment (jpichon) [inline README.md:52]: Should docs-requirements.txt be mentioned here or in the next section?

**Review by jpichon — CHANGES_REQUESTED** [review]
I think the branch name for CI is wrong, lgtm otherwise

File: .github/workflows/publish.yml (hunk 1 above)
Comment (jguiditta) [reply .github/workflows/publish.yml:5]: Good catch, forgot this is still using the old naming, will update today

File: README.md (hunk 2 above)
Comment (jguiditta) [reply README.md:52]: Sure, I can add that

**Review by jpichon — APPROVED** [review]

**Pull request merged**